htdocs/product/class/product.class.php

```diff
@@ -3518,20 +3518,26 @@ function getLibFinished()
 	 * 	@param		string	$label			Label of stock movement
 	 * 	@param		double	$price			Unit price HT of product, used to calculate average weighted price (PMP in french). If 0, average weighted price is not changed.
 	 *  @param		string	$inventorycode	Inventory code
+	 *  @param  	string	$origin_element Origin element type
+	 *  @param  	int		$origin_id      Origin id of element
 	 * 	@return     int     				<0 if KO, >0 if OK
 	 */
-	function correct_stock($user, $id_entrepot, $nbpiece, $movement, $label='', $price=0, $inventorycode='')
+	function correct_stock($user, $id_entrepot, $nbpiece, $movement, $label='', $price=0, $inventorycode='', $origin_element='', $origin_id=null)
 	{
 		if ($id_entrepot)
 		{
 			$this->db->begin();
 
 			require_once DOL_DOCUMENT_ROOT .'/product/stock/class/mouvementstock.class.php';
+			
+			// If remove stock then save it the actual PMP as price (the new calculation of PMP is triggered only with $movement = 0 || 3 )
+			if ($movement == 1) $price = $this->pmp;
```

Comment:
Still don't understand that.
If movement is 1, it means "remove". In such a case, the pmp won't be modified by dolibarr so no reason to save it.
And if you remove a negative quantity, it means it is an increase. PMP must be provided when calling function or set to 0 to keep it unchanged.